frame_transformer/src/frame_transformer_node.cpp

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2021 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+#include "frame_transformer/frame_transformer_node.hpp"
+#include <boost/algorithm/string/trim.hpp>
+
+namespace frame_transformer
+{
+  namespace std_ph = std::placeholders;
+
+  Node::Node(const rclcpp::NodeOptions &options)
+      : carma_ros2_utils::CarmaLifecycleNode(options)
+  {
+  }
+
+  std::unique_ptr<TransformerBase> Node::build_transformer() {
+    
+    // NOTE: IMU messages could be added once imu_pipeline package is ported to noetic and added as a dependency
+
+    auto type = config_.message_type;
+    boost::algorithm::trim(type); // Trim white space from the type string
+
+    if (type == "geometry_msgs/Vector3Stamped")
```

Comment:
Not a required change, but switch case would be quicker here

Reply:
switch cannot be used in C++ on strings